sites/hashdev/src/_pages/blog/4_announcement-error-stack.mdx

```diff
@@ -0,0 +1,283 @@
+
+This will eventually output:
+
+```plain
+Could not parse configuration file
+             at main.rs:6:10
+      - Could not read file "config.json"
+      - 1 additional opaque attachment
+
+Caused by:
+   0: No such file or directory (os error 2)
+             at main.rs:5:10
+
+Stack backtrace:
+   0: error_stack::report::Report<T>::new
+             at error-stack/src/report.rs:187:18
+   1: error_stack::context::<impl core::convert::From<C> for error::report::Report<C>>::from
+             at error-stack/src/context.rs:87:9
+   2: <core::result::Result<T,E> as error::result::IntoReport>::report
+             at error-stack/src/result.rs:204:31
+   3: parse_config
+             at main.rs:4:19
+   4: main
+             at main.rs:13:26
+   5: ...
+
+Suggestion: Use a file you can read next time!
+```
+
+## What's the catch?
+
+You might think that this comes with a lot of performance overhead, but with speed being a key concern of hEngine, we've taken this into account. The impact on the ‘happy' success code path is minimal; a `Report` is only one pointer in size and requesting from a `Report` only happens at the time of error reporting. We use a similar data layout as `anyhow` and thus expect similar performance.
+
+However, we've purposefully added in some requirements that result in a little bit of extra upfront **development** overhead. We think having explicitly typed `Result` values is worth requiring, and especially helpful in identifying different types across module/crate boundaries, so we still want users to create types (`Context`s). This approach makes `error-stack` usable not only for binaries but also for libraries. In most cases, the caller of a library only cares about the error directly returned from a library, not an underlying OS error. As the current context is _always_ known, the user directly has all type information without optimistically trying to downcast to an error type (which remains possible). This also implies that **more times than not** a developer is _forced_ to add a new context because the type system requires it. We've found from experience that this automatically greatly improves error reporting.
+
+We haven't made it easy to create a new `Report` from a `String` because it requires the type to implement `Context`. It's still possible to use a newtype `UniqueError(String)` if a user wants to, but we discourage this usage because we've found that a lot of those places benefit from attaching a string message on a well-defined context instead. In any case, other libraries most probably still have to change the context from `UniqueError` to their context because of the explicit type requirement.
+
+**By consciously imposing more restrictions in some areas we hope to reduce the _overall_ cost of development, in keeping with the philosophy underpinning Rust itself.**
+
+## Some other cool things about `error-stack` that we don't have time for!
```

Comment:
```suggestion
### Some other cool things about `error-stack`
```

We say "we don't have space" below, and saying "we don't have time" could be interpreted the wrong way by someone only scanning the headings.